src/java.base/share/classes/java/util/stream/AbstractPipeline.java

```diff
@@ -433,6 +433,21 @@ protected final boolean hasAnyStateful() {
          return result;
      }
 
+    /**
+     * Returns whether any of the stages in the (entire) pipeline is short-circuiting
+     * or not.
+     * @return {@code true} if any stage in this pipeline is stateful,
+     *         {@code false} if not.
+     */
+    protected final boolean isShortCircuitingPipeline() {
```

Comment:
protected can be replaced by package-private

Reply:
Since `hasAnyStateful()` is protected, I think this too should be, for consistency.